File: Internat1/mainframe.cpp
#include "mainframe.h"

#include "main.h"
#include "aboutlogo.h"
#include "appversion.h"

#include <wx/aboutdlg.h>

#ifndef __WXMSW__
#include "wxwin32x32.xpm"
#endif // __WXMSW__

MainFrame::MainFrame(const wxString& title) : wxFrame(NULL, -1, title)
{
#ifdef __WXDEBUG__
	wxPrintf(_T("Creating a \"MainFrame\" object\n"));
#endif // __WXDEBUG__

	SetIcon(wxICON(appIcon)); // Defining app icon

	CreateControls();

	ConnectControls();
}

MainFrame::~MainFrame()
{
#ifdef __WXDEBUG__
	wxPrintf(_T("Destroying a \"MainFrame\" object\n"));
#endif // __WXDEBUG__
}

void MainFrame::CreateControls()
{
	// Status bar
	CreateStatusBar(1);
	wxString sTxt=wxGetApp().GetBuildInfos(true);
	SetStatusText(sTxt);

	// Menu bar
	wxMenuBar *menuBar = new wxMenuBar();
	// File menu
	wxMenu *fileMenu = new wxMenu();
	fileMenu->Append(wxID_EXIT);
	menuBar->Append(fileMenu, wxGetStockLabel(wxID_FILE));
	// Help menu
	wxMenu* helpMenu = new wxMenu();
	helpMenu->Append(wxID_ABOUT);
	menuBar->Append(helpMenu, wxGetStockLabel(wxID_HELP));

	SetMenuBar(menuBar);

	// Add a panel to the frame (needed under Windows to have a nice frame)
	wxPanel *pnl=new wxPanel(this, -1);

	wxBoxSizer *sizer = new wxBoxSizer(wxVERTICAL);
		sizer->AddStretchSpacer(1);
		wxStaticText *label = new wxStaticText(pnl, wxID_STATIC, _("Welcome from wxWidgets"));
			wxFont fnt=label->GetFont().MakeLarger().MakeLarger();
			label->SetFont(fnt);
		sizer->Add(label, 0, wxALL|wxALIGN_CENTER, 10);
		sizer->AddStretchSpacer(1);
	pnl->SetSizer(sizer);
}

void MainFrame::ConnectControls()
{
	Bind(wxEVT_MENU, &MainFrame::OnMenuExitClicked, this, wxID_EXIT);
	Bind(wxEVT_MENU, &MainFrame::OnMenuAboutClicked, this, wxID_ABOUT);
}

void MainFrame::OnMenuExitClicked(wxCommandEvent& event)
{
	Close();
}

void MainFrame::OnMenuAboutClicked(wxCommandEvent& event)
{
	wxAboutDialogInfo info;
	info.SetName(_(PRODUCTNAME));
	info.SetVersion(wxString::Format(_T("(v%d.%d.%d)"), VERSION_MAJOR, VERSION_MINOR, VERSION_REV));
	info.SetDescription(_(DESCRIPTION));
	info.SetCopyright(COPYRIGHT);
	wxIcon ico;
	ico.CopyFromBitmap(wxGet_wxWidgets_png_Bitmap());
	info.SetIcon(ico);

	wxAboutBox(info);
}

File: Internat1/Python/MainFrame.py
import wx
import wx.adv

_ = wx.GetTranslation

class MainFrame(wx.Frame):

    def __init__(self):
        wx.Frame.__init__(self, None, title=wx.GetApp().GetAppName())
        self.SetIcon(wx.Icon('../../common/wxwin.ico'))

        self._createControls()

        self._connectControls()

    def _createControls(self):
        # A Statusbar in the bottom of the window
        self.CreateStatusBar(1)
        sMsg = 'wxPython ' + wx.version()
        self.SetStatusText(sMsg)

        # Menu bar
        menubar = wx.MenuBar()
        # File menu
        filemenu = wx.Menu()
        filemenu.Append(wx.ID_EXIT)
        menubar.Append(filemenu, wx.GetStockLabel(wx.ID_FILE))
        # Help menu
        helpmenu = wx.Menu()
        helpmenu.Append(wx.ID_ABOUT)
        menubar.Append(helpmenu, wx.GetStockLabel(wx.ID_HELP))

        self.SetMenuBar(menubar)

        # Add a panel to the frame (needed under Windows to have a nice background)
        pnl = wx.Panel(self, wx.ID_ANY)

        szrMain = wx.BoxSizer(wx.VERTICAL)
        szrMain.AddStretchSpacer(1)
        label = wx.StaticText(pnl, wx.ID_STATIC, _("Welcome from wxWidgets"))
        fnt = label.GetFont().MakeLarger().MakeLarger()
        label.SetFont(fnt)
        szrMain.Add(label, 0, wx.ALL|wx.ALIGN_CENTER, 10)
        szrMain.AddStretchSpacer(1)
        pnl.SetSizer(szrMain)

    def _connectControls(self):
        self.Bind(wx.EVT_MENU, self.onMenuExitClicked, id=wx.ID_EXIT)
        self.Bind(wx.EVT_MENU, self.onMenuAboutClicked, id=wx.ID_ABOUT)

    def onMenuExitClicked(self, evt):
        self.Close()

    def onMenuAboutClicked(self, evt):
        info = wx.adv.AboutDialogInfo()
        info.SetName('Internat')
        info.SetVersion('(v1.0.0)')
        info.SetDescription(_('wxWidgets I18N sample'))
        info.SetCopyright(_('Copyright (c) X.P. 2018'))
        info.SetIcon(wx.Icon('../../common/wxWidgets.png', wx.BITMAP_TYPE_PNG))

        wx.adv.AboutBox(info)
